### PR TITLE
Makes hiveminds sprite dependent on the type of weed theyre on + new hivemind sprites

### PR DESCRIPTION
**Pull request by ClosetedSkeletonGH**
## About The Pull Request
**Visual changes:**

**All new sprites/anims:**
image

-If hivemind moves onto normal weed/no weeds, it becomes a standard colored hivemind

-If hivemind moves onto sticky weeds, it becomes a sticky hivemind

-If hivemind moves onto resting weeds, it becomes a resting hivemind

-Hivemind remanifesting no longer looks buggy by hivemind moving to face the previous direction post manifesting 
-Hivemind nolonger flickers on demanifestation/remanifestation

Video of visual changes:

https://github.com/tgstation/TerraGov-Marine-Corps/assets/120620754/614fd526-5e43-45a2-8d2d-130ff55a5bcd

(ignore the sliding post manifestation in this vid, ive fixed that)

https://github.com/tgstation/TerraGov-Marine-Corps/assets/120620754/2b4f0634-6834-4962-af33-76cdd0d039a1

https://github.com/tgstation/TerraGov-Marine-Corps/assets/120620754/0abc38ba-8550-468c-bd71-8965353cf24c

**Gameplay changes:**

-Hivemind has had its fast and slow manifestation/demanifestation merged into one because slow demanifestation was flat out ugly and fast tele was too fast. Now tele's are just a standard speed of 1.1 (and 1 second movement prevention)

(this means that hivemind normal demanifestation and remanifestation is slightly slightly faster, but hivemind double click tele and minimap tele is a bit slower, barely noticeable besides looking better)

## Why It's Good For The Game

General sprite improvements that makes the hivemind look significantly cooler, also old hivemind sprite was just 3 colors total, which wasnt bad at all but it was meant for old weeds, not our current weeds
## Changelog
:cl:
add: Hivemind sprite changes depending on the type of weed it is on
balance: Hivemind demanifestation and remanifestation
fix: Hivemind no longer flickers on tele
fix: Hivemind direction of manifestation is normalized

/:cl:

### CODE REVIEW

**Comment by coldironwarrior** [comment]
finally...it's here

**Comment by aarrisu** [comment]
someone post awoos dog in this pr

**Comment by lbnesquik** [comment]
Very nice

**Review by Kuro020 — APPROVED** [review]
based

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 1]
```diff
@@ -242,12 +247,12 @@
 		return
 	abstract_move(get_turf(xeno))
 
-/// Hivemind just doesn't have any icons to update, disabled for now
+// handles hivemind updating with their respective weedtype
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:250]: This isn't autodoc'ed anymore (it needs 3 slashes)

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 2]
```diff
@@ -188,17 +190,20 @@
...
+	flick("Hivemind_[initial(loc_weeds_type.color_variant)]_materialisation_reverse", src)
+	setDir(SOUTH)
+	addtimer(CALLBACK(src, PROC_REF(end_teleport), T), TIME_TO_TRANSFORM)
 
 ///Finish the teleportation process to send the hivemind manifestation to the selected turf
 /mob/living/carbon/xenomorph/hivemind/proc/end_teleport(turf/T)
-	flick("Hivemind_materialisation_fast", src)
 	if(!check_weeds(T, TRUE))
 		balloon_alert(src, "No weeds in destination")
+		return
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:202]: If you're gonna early return here, you don't need the else beneath it

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 3]
```diff
@@ -449,7 +449,16 @@
 /mob/living/carbon/xenomorph/proc/handle_weeds_on_movement(datum/source)
 	SIGNAL_HANDLER
 	var/obj/alien/weeds/found_weed = locate(/obj/alien/weeds) in loc
+	if(loc_weeds_type == found_weed?.type)
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm:452]: You should check !found_weed on top of what you've got here, instead of using ?

**Review by Xander3359 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 4]
```diff
@@ -135,7 +136,7 @@
 		set_datum(FALSE)
 		hive.xenos_by_upgrade[upgrade] += src
 		update_wounds()
-		update_icon()
+		update_icon_state()
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:139]:
```suggestion
		update_icon()
```
You're only supposed to change it in the proc that changes the icon states, not where you're calling it. Update_icon is meant to handle calling update_icon_state for you.

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 5]
```diff
@@ -242,12 +246,12 @@
 		return
 	abstract_move(get_turf(xeno))
 
-/// Hivemind just doesn't have any icons to update, disabled for now
-/mob/living/carbon/xenomorph/hivemind/update_icon()
+/// handles hivemind updating with their respective weedtype
+/mob/living/carbon/xenomorph/hivemind/update_icon_state()
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:250]: This is the only place that needed to be changed.

**Review by Xander3359 — APPROVED** [review]

**Comment by chizzy376** [comment]
Suggestion for dirt
poopmind

**Comment by yyzsong** [comment]
Floating purple eye attack

File: code/game/objects/effects/weeds.dm [hunk 6]
```diff
@@ -1,5 +1,5 @@
 //Color variant defines
-#define SPEED_COLOR ""
+#define NORMAL_COLOR ""
```
Comment (Lumipharon) [inline code/game/objects/effects/weeds.dm:2]: this should really be like, an actual define, not just a blank. Not sure if you'll need to update anything else to take this into account though.

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 7]
```diff
@@ -445,11 +445,26 @@
...
 
-/// Handles logic for the xeno moving to a new weeds tile
+/**  Handles logic for the xeno moving to a new weeds tile.
+Returns TRUE when loc_weeds_type changes. Returns FALSE when it doesn’t change */
 /mob/living/carbon/xenomorph/proc/handle_weeds_on_movement(datum/source)
 	SIGNAL_HANDLER
 	var/obj/alien/weeds/found_weed = locate(/obj/alien/weeds) in loc
-	loc_weeds_type = found_weed?.type
+	if(isnull(found_weed))
+		if(!(isnull(loc_weeds_type)))
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm:454]: why are you doing null checks specifically?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 8]
```diff
@@ -242,12 +246,12 @@
...
 	if(status_flags & INCORPOREAL)
 		icon_state = "hivemind_marker"
 		return
-	icon_state = "Hivemind"
+	icon_state = "Hivemind_[initial(loc_weeds_type.color_variant)]"
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:254]: is this not erroring? loc_weeds_type could be null, couldn't it?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Review by Lumipharon — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 9]
```diff
@@ -445,11 +445,27 @@
...
+	. = ..()
+	if(!.)
+		return
+	update_icon()
+
+/mob/living/carbon/xenomorph/hivemind/handle_weeds_on_movement(datum/source)
+	. = ..()
+	if(!.)
+		return
+	update_icon()
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm:468]: Why is this duplicated?

**Pull request merged**